### PR TITLE
Add `handwired/dactyl_lightcycle` keyboard

### PR DESCRIPTION
**Pull request by dunk2k**
## Description

- `default` and `via` keymaps
- support for QMK layouts:
    - split_3x6_3
    - split_3x5_3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_lightcycle/readme.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Dactyl Lightcycle
+
+![TopElev](https://pbs.twimg.com/media/C_lE5BUU0AEufDT?format=jpg)
```
Comment (drashna) [inline keyboards/handwired/dactyl_lightcycle/readme.md:3]: this should be rehosted on imgur or the like.

Comment (dunk2k) [reply keyboards/handwired/dactyl_lightcycle/readme.md:3]: amended as requested

File: keyboards/handwired/dactyl_lightcycle/readme.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Dactyl Lightcycle
+
+![TopElev](https://i.imgur.com/MwPhTrC.jpg)
```
Comment (drashna) [inline keyboards/handwired/dactyl_lightcycle/readme.md:3]:
```suggestion
![TopElev](https://i.imgur.com/MwPhTrCh.jpg)
```

File: keyboards/handwired/dactyl_lightcycle/config.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/* mouse config */
+#define MOUSEKEY_DELAY          0
+// Lowering MOUSEKEY_INTERVAL raises the cursor speed significantly, you may want to lower MOUSEKEY_MAX_SPEED (default: 10)
+#define MOUSEKEY_INTERVAL       20
+#define MOUSEKEY_MAX_SPEED       7
+#define MOUSEKEY_WHEEL_DELAY 0
+
+/* layer config */
+#define TAPPING_TOGGLE 1
+
```
Comment (drashna) [inline keyboards/handwired/dactyl_lightcycle/config.h:29]: Generally, this is stuff we want defined at the keymap level, not the keyboard level.

Comment (dunk2k) [reply keyboards/handwired/dactyl_lightcycle/config.h]: `Config.h` contents split - do amendments meet request?

File: keyboards/handwired/dactyl_lightcycle/info.json [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        "cols": ["D4", "C6", "D7", "E6", "B4", "B5"],
+        "rows": ["F7", "B1", "B3", "B2", "B6"]
+    },
+    "diode_direction": "COL2ROW",
+    "split": {
+        "enabled": true,
+        "soft_serial_pin": "D0"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/handwired/dactyl_lightcycle/info.json:35]: If the dev board is set, these don't need to be.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/dactyl_lightcycle/readme.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+A handwired, split bodied, concave key-well, columnar keyboard.  
+The *Dactyl Lightcycle* is a variation of the [Dactyl](/keyboards/handwired/dactyl/) keyboard, its differences are:
+1. Top (number) row removed
+2. Five key thumb cluster; one less than the *Dactyl's* six
+
+This modernised rendition of the *Dactyl Lightcycle* requires two Pro Micro controllers, rather than the *Dactyl's* original implementation of a micro controller and I/O expander.
+
+Keyboard Maintainer: [Matt Aderth](https://github.com/adereth/)  
+Hardware Supported: 2x Pro Micro micro controller, or clone of  
+Hardware Availability: Case files available on [GitHub](https://github.com/adereth/dactyl-keyboard), file names beginning "lightcycle" apply
```
Comment (fauxpark) [inline keyboards/handwired/dactyl_lightcycle/readme.md:14]:
```suggestion
* Keyboard Maintainer: [Matt Aderth](https://github.com/adereth/)
* Hardware Supported: 2x Pro Micro micro controller, or clone of
* Hardware Availability: Case files available on [GitHub](https://github.com/adereth/dactyl-keyboard), file names beginning "lightcycle" apply
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request merged**